### ISSUE TITLE
[Combat Report] Balance in Pre-Early Access - Part 1

### ISSUE THREAD

**Issue by SabreMania** [tsunamayo/Starship-EVO#1861]
Greetings Tsuna,

Though I have many, many things I could talk about, I will (try to) stick to something brief with not too many details, as, from what I could gather, this is the kind of reports that you like. Understandably so, too, given that you need to code the entire game on your own. Nevertheless, here is my first dedicated battle report post. Enjoy.

This report is based on an uncountable number of skirimishes between my Half-Baked destroyer, and dozens of the Corellian Shuttle, as well as two separate battles against the Whirlwind. 

Keep in mind, even in the latest version, I still have the bug that allows my guns to fire at an absurd fire rate, but we are running out of time, early access is in a few weeks, and my building pace is too slow, so I had to make due with what I had. 

So, for starters. I would like to comment on beams. It's no secret that they are pretty strong, game breaking strong, I would say, but where they get even more broken is when you use them in their brick form. I was told that you were aware that they did not consume as much heat as they should, but even then, the damage they do is ridiculous, and even with three times as much heat consumption as they currently have, they would still be a go-to weapon. 

![AAgun2](https://user-images.githubusercontent.com/52141971/80286393-2f7e3380-86f9-11ea-9f0b-c2af80cdaabc.png)

This little turret is 3 blocks wide, and one block and a quarter block high. It inflicts 18,000 heat damage over 1.5 seconds to it's target, as well as 1890 hull damage in the same time (assuming that Firing time really is Firing time, and not time to reload) , and I have 7 of them on my ship. 5 of them can fire at the same target under optimal conditions. For small weapons, there is no alternative. Beams outclass everything else. I cannot comment on large beams, as I have not tested them yet.

Next I would like to comment on shields. Against the small shuttles, the 5 second 'reload' for shields is... annoying, and would need some tweaking, but overall I don't think it's -too- broken, if we compare it to a large ship. I could down them easily and deal the damage that needed to be done, and overall the battles lasted 30 seconds, top.

Where it gets tricky is in the two battles I had against the Whirlwind, and that is where I realized just how much tweaking is going to be needed. The battles were not battles as much as drawn out executions. The Whirlwind is an easily outclassed opponement, for it's size, and in neither of the fights did it manage to get my shields below 3/4 of their maximum capacity. My Destroyer is roughly the same length, though it is quite a bit beefier, if a little thinner, so it's no surprise that it can handle much more systems.

The Whirlwind however did have something very, very annoying. It's shields recharge completely within 5 seconds of it's depletion, and shields in general start recharging after such an amount of time if they have not been shot. This is far, far too quick. This combined with the fact that I constantly had to chase it because it kept running away made it very tedious to get the kill. I would shoot at it, down the shields in about 10 seconds (assuming it rushed me and didn't try to avoid my full broadside), deal a little bit of damage, and 5 seconds later, I had to repeat the process.

Even with the repeating guns bug, this was very long. I cannot imagine what it would've been like if I would've had a normal fire rate. 

![whirl1](https://user-images.githubusercontent.com/52141971/80286765-4160d600-86fb-11ea-9c24-1aeb15bced27.png)
![whirl2](https://user-images.githubusercontent.com/52141971/80286768-42920300-86fb-11ea-8511-4b6fee7f3f78.png)
![whirl3](https://user-images.githubusercontent.com/52141971/80286769-43c33000-86fb-11ea-8ff9-aa19aa81f9c5.png)

This is the damage it took to destroy the ship. I will say that it was a health kill, not a heat kill. Due to how strong and how quick shields recharge, I cannot comment on how good or bad the Heat mechanic is. 

So this was my first battle report. If you would like to ask questions or details that I forgot to mention (this post was quite a bit longer than I wanted it to be, even as a 'shrunk down' version), feel free to ask. On another note, here is a picture of my unfinished destroyer, to give you an idea of it's firepower. Some of the turrets are missing, as they dissapeared during the fight, and not due to damage taken.

![destroyer2](https://user-images.githubusercontent.com/52141971/80286871-fb584200-86fb-11ea-9322-19da7051b850.png)

**Comment by tsunamayo** [comment]
Thanks you very much for this report!
Okay just so I can know better, what is the size and mass of your ship?
You were able to put 7 * 4 * fulls tack of little beam on your ship right?
Also people seems to not be using capacitor for shield - the recharge rate is super short but that is by design. Once you put more capacitor it will become longer. That being said shield require some more tuning, like you said the hit cooldown, and maybe slow the recharge rate.
Will do a more detailed answer soon!

**Comment by SabreMania** [comment]
Ah sorry, I thought I had screenshotted it, but I guess I missed that part.

Keep in mind, the ship isn't finished. The exterior needs a lot of work, and there is no interior at all, except for rushed systems because I wanted to be able to test combat roughly, as soon as possible. But regardless.

The mass is 4395.5t, the size is 27x50x123, with a volume of 166 050m3

I was able to put all of these beams on my ship yes, it did not raise my heat in the slightest. Either the heat bar progression is bugged, or it simply did not affect my ship. I will say however, that my turrets bugged out like I mentionned earlier in the post, so in practice I only had 3 of those miniturrets.

And yes, people don't use capacitors for shields. In fact, I didn't use any capacitors in my build, be it for shields or for the energy generation. However I can't really comment on them, as I never used them.

I will be doing another report soon, will be testing my ship against different variants of itself, in order to have a more fair fight, since the Whirlwind is just too weak for it. Would you prefer that I post the result in here, or in another, new thread?

**Comment by ProPeach** [comment]
If we're supposed to use capacitors to increase the shield hp, how about reducing the hp that that shield block adds to the shield? Right now the shield block adds enough shield hp that you don't need a capacitor so players aren't using them, but if shield blocks produced weak shields by themselves the capacitors would be much more useful

**Comment by tsunamayo** [comment]
> If we're supposed to use capacitors to increase the shield hp, how about reducing the hp that that shield block adds to the shield? Right now the shield block adds enough shield hp that you don't need a capacitor so players aren't using them, but if shield blocks produced weak shields by themselves the capacitors would be much more useful

Yes I will do exactly that. I will make both capacity and regen weaker, maybe not in proportion. But it means that the time to recharge might be even shorter!

**Comment by ProPeach** [comment]
> But it means that the time to recharge might be even shorter!

The shortest possible regen time can be chosen to be anything you like - if one shield block can fill up it's own shield hp in say 5 seconds, that time should be the same whether you have 10 shield blocks or 1 if the shield hp and recharge rate scale linearly.
The only way that more shield blocks could regen faster than fewer shield blocks is if the regen rate scaled at a larger rate than the shield max hp. In my opinion the opposite would actually work better - Scale shield hp of a shield block linearly, but give a lesser gain in shield recharge rate the larger you stretch a shield block. 
For example
1 shield block - 100 hp, charge in 5 seconds - 20hp/s
2 shield block - 200 hp, charge in 5.2 seconds - 38hp/s
3 shield block - 300 hp, charge in 5.4 seconds - 55 hp/s etc
This way a larger shield block can still charge more capacitors faster, but it doesn't increase the minimum recharge time of 5 seconds

**Comment by tsunamayo** [comment]
@ProPeach no because it is an incentive to spam many small unit instead of making larger one. Bad for performance - I prefer people making big stack. It just that with a super base hp the rate is always gonna be short, so I have to make clear that people needs to use battery more. Will do in next patch!

**Comment by SabreMania** [comment]
So wait, having small and numerous 'packs' of systems is less efficient than having a single big slab of system? How come? And isn't having capacitors already a way to encourage such small systems? You can't put capacitors on blocks that are behind other blocks after all

**Comment by tsunamayo** [comment]
@SabreMania yes true, but for now they have the same efficiency. What @ProPeach proposed was a negative effiency, aka putting smaller stack should be more efficient, which I disagreed.
